### PR TITLE
Improve test coverage - Add unit tests

### LINKED ISSUE #159: Improve test coverage
- [ ] Logic Manager test: add commands
- [ ] Update testUtils to support task with time
- [ ] guiTests for rename, schedule, mark, unmark, undo (maybe not all)
- [ ] More unit test?

### PR DESCRIPTION
**Pull request by INCENDE**
Fixes #159

- [x] MainApp

Commons:
- [x] UnmodifiableObservableList

Logic:
- [x] Add command
- [x] DateTimeUtils
- [x] IncorrectCommand
- [x] Parser
- [x] Alias/Unalias

Model:
- [x] UserPrefs
- [x] UniqueTaskList
- [x] Name
- [x] ToDoList
- [x] Task

UI:
- [ ] UiPartLoader
- [ ] UiManager
- [ ] CompletedTasksPanel

### CODE REVIEW

**Comment by burnflare** [comment]
Let's use this PR to collect more

File: src/test/java/seedu/agendum/storage/XmlAdaptedTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        
+        Task taskWithNoEndDateTime = new Task(new Name("taskWithNoEndDateTime"), optionalStartDateTime, Optional.ofNullable(null));
+        taskWithNoEndDateTime.markAsCompleted();
+        xmlAdaptedTaskNoEndDateTime = new XmlAdaptedTask(taskWithNoEndDateTime);
+    }
+    
+    public void assertTaskEqual(Task task, Optional<LocalDateTime> startDateTime, Optional<LocalDateTime> endDateTime, boolean isCompleted) {
+        assertTrue(task.getStartDateTime().equals(startDateTime));
+        assertTrue(task.getEndDateTime().equals(endDateTime));
+        assertTrue(task.isCompleted() == isCompleted);
```
Comment (rachx) [inline src/test/java/seedu/agendum/storage/XmlAdaptedTaskTest.java]: oops most probably my missing bracket from somewhere. fix it and the source?

**Comment by rachx** [comment]
@INCENDE Can suggest some unit tests to write from other components such as Logic and Model? We will do it

**Comment by burnflare** [comment]
+1%!! 🎉
Let's collect till +8% in this PR before merging it?

**Comment by INCENDE** [comment]
Model:

- [ ] Add command
- [ ] 
- [ ] 
- [ ]

**Comment by burnflare** [comment]
@INCENDE I think better to move the TODO list to the description of the PR. Feel free to delete this comment after moved.

File: src/test/java/seedu/agendum/storage/XmlAdaptedTaskTest.java (hunk 1 above)
Comment (rachx) [inline src/test/java/seedu/agendum/storage/XmlAdaptedTaskTest.java]: my bad. i thought the other isCompleted is another task

**Comment by rachx** [comment]
Should we collect gui tests and integration tests here or separately?

**Comment by burnflare** [comment]
Let's just collect everything here since there's a low chance of conflict with master for test classes.

**Comment by INCENDE** [comment]
89.9% !! Almost there

**Comment by rachx** [comment]
Make by, from and to case insensitive
#169

**Comment by rachx** [comment]
Those under logic are tested and covered by logic manager test.
Might want to add gui tests instead

**Comment by INCENDE** [comment]
Reached 91%! 
Also added new To-Dos

**Review by burnflare — APPROVED** [review]
Quite a lot of commits here already. Fix those merge conflicts and merge in!

Further changes can be done in another PR

**Pull request merged**